llmfoundry/models/layers/cross_entropy_loss.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Copied from https://github.com/Dao-AILab/flash-attention/blob/v1.0.9/flash_attn/losses/cross_entropy.py
```

Comment:
Should we use latest commit before triton refactor:
https://github.com/Dao-AILab/flash-attention/blob/f1a73d074002226c42ce65a1df170ecff9f022c0/flash_attn/losses/cross_entropy.py

Also use [permanent link](https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files)
for v1.0.9 link is: https://github.com/Dao-AILab/flash-attention/blob/6d48e14a6c2f551db96f0badc658a6279a929df3/flash_attn/losses/cross_entropy.py
for latest commit before triton: https://github.com/Dao-AILab/flash-attention/blob/f1a73d074002226c42ce65a1df170ecff9f022c0/flash_attn/losses/cross_entropy.py

Reply:
Done.